Invalid HTTP responses for custom languages are cached indefinitely

**Describe the bug**

There's an issue in the logic at: 
https://github.com/naptha/tesseract.js/blob/adcb5b875986ebe61d7235e64d956d648c53c1dd/src/worker-script/index.js#L101-L104

It doesn't check if the response is successful, which results in a bad response being cached indefinitely. This can happen if the server is down or there's a temporary 404 error, resulting in clients being broken until the IndexedDB is cleared.

This results in the following error being logged on all subsequent usages even if the server error was resolved:

```
tesseract-core.wasm.js:9 Error opening data file ./mrza.traineddata
tesseract-core.wasm.js:9 Please make sure the TESSDATA_PREFIX environment variable is set to your "tessdata" directory.
tesseract-core.wasm.js:9 Failed loading language 'mrza'
tesseract-core.wasm.js:9 Tesseract couldn't load any languages!
```

**To Reproduce**

Use an invalid language and notice that the response is cached in IndexedDb.

**Expected behavior**

An error should be thrown.

**Screenshots**

In the screenshot below, `mrza` is a file that didn't exist, notice the length of the Unit8Array. It's a 404 response.

<img width="951" alt="Screenshot 2021-11-25 at 18 37 14" src="https://user-images.githubusercontent.com/697707/143477484-4a667b27-08e6-4ee9-8f54-bdfd9a960ef4.png">

**Additional context**

I'll open a PR with a fix shortly.

--- a/src/worker-script/index.js
+++ b/src/worker-script/index.js
@@ -99,7 +99,11 @@ const loadLanguage = async ({
         }
 
         if (path !== null) {
-          const resp = await (isWebWorker ? fetch : adapter.fetch)(`${path}/${lang}.traineddata${gzip ? '.gz' : ''}`);
+          const fetchUrl = `${path}/${lang}.traineddata${gzip ? '.gz' : ''}`;
+          const resp = await (isWebWorker ? fetch : adapter.fetch)(fetchUrl);
+          if (!resp.ok) {
+            throw Error(`Network error while fetching ${fetchUrl}. Response code: ${resp.status}`)
+          }
           data = await resp.arrayBuffer();
         } else {
           data = await adapter.readCache(`${langPath}/${lang}.traineddata${gzip ? '.gz' : ''}`);

